### PR TITLE
Use index image for the dev flow

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Use the index image with the database generated via `opm` to deploy the cluster for development usage.

It introduces a lot of dependencies under our vendor directory that not really related to the PAO, so probably I can revert to use docker image instead of building the `opm` under the system.

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
@fromanirh FYI

File: Makefile [hunk 1]
```diff
@@ -117,9 +117,9 @@ registry-container: generate-manifests-tree
 	$(IMAGE_BUILD_CMD) build --no-cache -f openshift-ci/Dockerfile.registry.upstream.dev -t "$(FULL_REGISTRY_IMAGE)" --build-arg FULL_OPERATOR_IMAGE="$(FULL_OPERATOR_IMAGE)"  .
 
 .PHONY: index-container
-index-container: generate-manifests-index
+index-container: generate-index-database
 	@echo "Building the performance-addon-operator index image"
-	$(IMAGE_BUILD_CMD) build --no-cache -f openshift-ci/Dockerfile.index.upstream.dev -t "$(FULL_INDEX_IMAGE)" .
+	$(IMAGE_BUILD_CMD) build --no-cache -f build/_output/index.Dockerfile -t "$(FULL_INDEX_IMAGE)" build/_output
```
Comment (fromanirh) [inline Makefile:122]: this `index.Dockerfile` comes from the `opm` tool, right?

**Review by fromanirh — COMMENTED** [review]
So this is not meant to be run inside the OCP CI, right?

File: Makefile (hunk 1 above)
Comment (cynepco3hahue) [reply Makefile:122]: yep exactly

Comment (cynepco3hahue) [reply Makefile:122]: nope, it not, for the OpenShift CI, we already have the relevant PRs merged.

**Comment by fromanirh** [comment]
/lgtm
thanks @cynepco3hahue

**Pull request merged**